docs/user-guide.md

```diff
@@ -48,114 +61,71 @@ Cygwin, and Msys (MinGW) shells, using version 4.9.1 of the GNU toolchain.
-    Tree (FDT) source files (`.dts` files) provided with this software.
+Install the required packages to build Trusted Firmware with the following
+command:
 
-*   For debugging, ARM [Development Studio 5 (DS-5)][DS-5] v5.22.
+    sudo apt-get install build-essential gcc make git
 
+Download and install the AArch64 little-endian GCC cross compiler as indicated
+in the [Linaro instructions][Linaro SW Instructions].
 
-4.  Getting the Trusted Firmware source code
---------------------------------------------
+In addition, the following optional packages and tools may be needed:
 
-The Trusted Firmware (TF) source code can be obtained as part of the standard
-Linaro releases, which provide a full software stack, including TF, normal
-world firmware, Linux kernel and device tree, file system as well as any
-additional micro-controller firmware required by the platform. This TF version
-is tested with the [Linaro 15.10 Release][Linaro Release Notes].
-
-Note 1: Both the LSK kernel or the latest tracking kernel can be used with TF;
-choose the one that best suits your needs.
-
-Note 2: Currently to run the latest tracking kernel on FVP with GICv3 driver,
-some modifications are required to UEFI. Refer
-[here](#11--changes-required-for-booting-linux-on-fvp-in-gicv3-mode)
-for more details.
-
-The TF source code will then be in `arm-tf/`. This is the upstream git
-repository cloned from GitHub. The revision checked out by the `repo` tool is
-indicated by the manifest file. Depending on the manifest file you're using,
-this might not be the latest upstream version. To synchronize your copy of the
-repository and get the latest updates, use the following commands:
+*   `device-tree-compiler` package if you need to rebuild the Flattened Device
+    Tree (FDT) source files (`.dts` files) provided with this software.
 
-    # Change to the Trusted Firmware directory.
-    cd arm-tf
+*   `libssl-dev` package if Trusted Board Boot is enabled in the build.
 
-    # Download the latest code from GitHub.
-    git fetch github
+*   For debugging, ARM [Development Studio 5 (DS-5)][DS-5].
 
-    # Update your working copy to the latest master.
-    # This command will create a local branch master that tracks the remote
-    # branch master from GitHub.
-    git checkout --track github/master
 
+4.  Getting the Trusted Firmware source code
+--------------------------------------------
 
-Alternatively, the TF source code can be separately cloned from the upstream
-GitHub repository:
+Download the Trusted Firmware source code from Github:
 
     git clone https://github.com/ARM-software/arm-trusted-firmware.git
 
 
 5.  Building the Trusted Firmware
 ---------------------------------
 
-To build the Trusted Firmware images, change to the root directory of the
-Trusted Firmware source tree and follow these steps:
+*   Before building Trusted Firmware, the environment variable `CROSS_COMPILE`
+    must point to the Linaro cross compiler:
```

Comment:
It would still be easy to not click through web pages to get the actual
toolchain. The previous documentation was better as it provided a direct
"wget" command.

On Thu, 14 Apr 2016 at 09:49 danh-arm <notifications@github.com> wrote:

> In docs/user-guide.md
> <https://github.com/ARM-software/arm-trusted-firmware/pull/594#discussion_r59751400>
> :
>
> >
> >      git clone https://github.com/ARM-software/arm-trusted-firmware.git
> >
> >
> >  5.  Building the Trusted Firmware
> >  ---------------------------------
> >
> > -To build the Trusted Firmware images, change to the root directory of the
> > -Trusted Firmware source tree and follow these steps:
> > +*   Before building Trusted Firmware, the environment variable `CROSS_COMPILE`
> > +    must point to the Linaro cross compiler:
>
> Further up this patch says:
> "This document assumes that the reader has previous experience running a
> fully
> bootable Linux software stack on Juno or FVP using the prebuilt binaries
> and
> filesystems provided by [Linaro][Linaro Release Notes]"
>
> which points to this link:
>
> https://community.arm.com/docs/DOC-10952#jive_content_id_Linaro_Release_1602
>
> That page point to the Linaro cross compiler (about 6 lines down). Is that
> not enough?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/ARM-software/arm-trusted-firmware/pull/594/files/ec0a975fe809b28de4a8f0d68d5c894e081760ba#r59751400>
>